designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/StateDefinitionDetails.scala

```diff
@@ -0,0 +1,24 @@
+package pl.touk.nussknacker.engine.api.deployment
+
+import java.net.URI
+
+/**
+  * It is used to specify:
+  * <ul>
+  * <li>fixed default properties of a status: icon, tooltip, descripition
+  * <li>fixed set of properties of filtering options: displayableName, icon tooltip
+  * </ul>
+  * When a status has dynamic properties use ProcessStateDefinitionManager to handle them.
+  *
+  * @see default values of a status in [[ProcessStateDefinitionManager]]
+  * @see filtering options in [[UIStateDefinition]]
+  * @see overriding state definitions in [[OverridingProcessStateDefinitionManager]]
+  */
+case class StateDefinitionDetails(displayableName: String,
+                                  icon: URI,
+                                  tooltip: Option[String],
+                                  description: Option[String])
```

Comment:
Hmm, why not add here allowedActions and remove ProblemStateStatus? After that change we won't need `statusActionsPF`

Reply:
Good point, but we still will need it in some cases because this definition is only for static cases - see the docs above